patterns/vsi-extension/config.tf

```diff
@@ -0,0 +1,90 @@
+##############################################################################
+# Create Pattern Dynamic Variables
+# > Values are created inside the `dynamic_modules/` module to allow them to
+#   be tested
+##############################################################################
+
+module "dynamic_values" {
```

Comment:
the `dynamic_values/config_modules/security_groups` is a module that can be called here on its own. If you can, you can call that directly here instead of the main `dynamic_values`. This will help, because many of the changes you have made to other dynamic values modules were only necessary because of the values in this special case, but if all you need is subnet config then you can call that directly and you will not have to alter all of the other modules.